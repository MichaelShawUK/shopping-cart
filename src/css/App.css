.App {
  flex: 1;
  position: relative;
  animation-fill-mode: forwards;
  animation-delay: 1s;
}

#home-main {
  color: white;
  background-color: rgba(255, 255, 255, 0.6);
  animation-duration: 1s;
  animation-timing-function: ease-in-out;
}

#slide-left {
  position: absolute;
  width: 0;
  bottom: 0;
  top: 0;
  right: 0;
  background-color: white;
  box-shadow: 0 0 5px black;
  animation-name: slideleft;
  animation-duration: 1s;
  animation-delay: 0s;
  animation-fill-mode: forwards;
  opacity: 0;
}

#slide-left::before {
  content: "Unlock your sound";
  position: absolute;
  width: min-content;
  text-align: center;
  font-size: 3rem;
  text-transform: uppercase;
  left: -180px;
  margin-top: 100px;
}

@keyframes slideleft {
  from {
    width: 0;
    opacity: 0;
  }
  to {
    width: 35%;
    opacity: 1;
  }
}

@keyframes slidedown {
  from {
    height: 0;
  }
  to {
    height: 93vh;
  }
}

@keyframes overlay {
  from {
    box-shadow: 0;
  }
  to {
    box-shadow: inset 0 0 0 2000px rgba(255, 255, 255, 0.6);
  }
}

@keyframes fadein {
  from {
    opacity: 0;
  }
  to {
    opacity: 1;
  }
}

@media (max-width: 600px) {
  .main-content {
    font-size: 2rem;
  }

  .flex-main {
    width: fit-content;
  }
}
