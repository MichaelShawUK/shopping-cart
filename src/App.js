import "./css/App.css";

const App = () => {
  return (
    <div className="App">
      <div id="home-main"></div>
      <div id="slide-left"></div>
    </div>
  );
};

export default App;
