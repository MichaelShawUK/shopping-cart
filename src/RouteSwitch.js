import { BrowserRouter, Routes, Route } from "react-router-dom";
import App from "./App";
import Shop from "./components/Shop";
import Cart from "./components/Cart";
import Nav from "./components/Nav";
import Product from "./components/Product";

const RouteSwitch = () => {
  return (
    <BrowserRouter>
      <Nav />
      <Routes>
        <Route path="/" element={<App />} />
        <Route path="/shop" element={<Shop />} />
        <Route path="/cart" element={<Cart />} />
        <Route path="/shop/:id" element={<Product />} />
      </Routes>
    </BrowserRouter>
  );
};

export default RouteSwitch;
