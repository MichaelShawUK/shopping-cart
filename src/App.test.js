import { render, screen } from "@testing-library/react";
import RouteSwitch from "./RouteSwitch.js";
import App from "./App";

test("renders nav", () => {
  render(<RouteSwitch />);
  const nav = screen.getByRole("navigation");
  expect(nav).toBeInTheDocument();
});

describe("App component", () => {
  it("matches snapshot", () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });
});
