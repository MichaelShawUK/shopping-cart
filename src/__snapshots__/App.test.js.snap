// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component matches snapshot 1`] = `
<div>
  <div
    class="App"
  >
    <div
      id="home-main"
    />
    <div
      id="slide-left"
    />
  </div>
</div>
`;
